frontend/src/concepts/k8s/NameDescriptionField.tsx

```diff
@@ -34,84 +25,97 @@ const NameDescriptionField: React.FC<NameDescriptionFieldProps> = ({
   disableK8sName,
 }) => {
   const autoSelectNameRef = React.useRef<HTMLInputElement | null>(null);
+  const [showHiddenResourceName, setShowHiddenResourceName] = React.useState<boolean | undefined>(
+    showK8sName,
+  );
+
+  const [validateK8sName, setValidateK8sName] = React.useState<K8sNameValidationState>({
+    ruleLength: 'indeterminate',
+    ruleCharacterTypes: 'indeterminate',
+  });
```

Comment:
We may need to include this in the state of the form variables to stop submit when these are in error (and any other condition we add)

Also, on edit, you don't manage it, and it looks weird:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/8126518/2f24941a-fad6-4ace-8fbb-c66753032832)

Reply:
sorry, I have missed the first comment here. I have updated the edit workbench validation.

Reply:
handled the first comment too.